File: src/collectors/sources/SegyeSourceAdapter.ts
import { BaseRSSAdapter } from "./BaseRSSAdapter.js";

/**
 * 세계일보 RSS 수집 어댑터.
 * 종합 일간지로, 경제·부동산·생활경제 기사를 제공합니다.
 */
export class SegyeSourceAdapter extends BaseRSSAdapter {
  constructor() {
    super({
      sourceName: "세계일보",
      sourceType: "news_media",
      feedUrls: [
        "https://www.segye.com/Articles/RSSList/segye_economy.xml",
      ],
    });
  }
}

File: src/main/java/com/economicbriefing/collector/source/SegyeSourceAdapter.java
package com.economicbriefing.collector.source;

import java.util.List;

import com.economicbriefing.collector.filter.CategoryClassifier;
import com.economicbriefing.collector.parser.ArticleNormalizer;
import com.economicbriefing.collector.parser.RssParser;
import com.economicbriefing.domain.article.ArticleSourceType;
import com.economicbriefing.domain.article.NewsCategory;
import org.springframework.stereotype.Component;

@Component
public class SegyeSourceAdapter extends AbstractRssSourceAdapter {

    public SegyeSourceAdapter(RssParser rssParser, ArticleNormalizer normalizer, CategoryClassifier categoryClassifier) {
        super(rssParser, normalizer, categoryClassifier);
    }

    @Override
    public String getSourceName() { return "세계일보"; }

    @Override
    public String getFeedUrl() { return "https://www.segye.com/Articles/RSSList/segye_economy.xml"; }

    @Override
    public ArticleSourceType getSourceType() { return ArticleSourceType.NEWS_MEDIA; }

    @Override
    public List<NewsCategory> getDefaultCategories() { return List.of(); }
}
